src/Polly.Core/CircuitBreaker/Controller/CircuitStateController.cs

```diff
@@ -16,6 +16,7 @@ internal sealed class CircuitStateController<T> : IDisposable
     private readonly ResilienceStrategyTelemetry _telemetry;
     private readonly CircuitBehavior _behavior;
     private readonly TimeSpan _breakDuration;
```

Comment:
```suggestion
    private readonly ValueTask<TimeSpan> _breakDurationProvider;
```

I suggest to Initialize `_breakDurationProvider` in the constructor to unify the paths and simplify the changes.

Reply:
Is this true though? I am pretty sure that CB is closed and only becomes open afterwards. 

The break duration provider is only called after the constructor is called.

Reply:
I apologize for the confusion earlier. There was a translation error in my previous reply.

Reply:
Looking at the code, you keep `TimeSpan _breakDuration` and just add a new field (as you have done).

It's only used at one place, so there won't be any repetition.